File: nowa/2015/python/zad4_2.py
# ladowanie danych
with open('../dane/liczby.txt') as f:
    data = []
    for line in f.readlines():
        data.append(line[:-1])

# sprawdzenie dzielnosci liczb
div2 = 0
div8 = 0
for line in data:
    # jezeli liczba konczy sie zerem, to znaczy ze jest podzielne przez 2
    if line[-1] == '0':
        div2 += 1
    # jezeli liczba konczy sie trzema zeremi, to znaczy ze jest podzielne przez 8
    if line[-3:] == '000':
        div8 += 1

# zapisanie odpowiedzi
answer = f'4.2. Liczby podzielne przez 2: {div2}; Liczby podzielne przez 8: {div8}'
print(answer)

File: nowa/2015/c++/zad4_2.cpp
#include <iostream>
#include <cstdlib>
#include <vector>
#include <string>
#include <fstream>
using namespace std;

string zad4_2()
{
    string line;
    vector<string> content;

    fstream file("../dane/liczby.txt");

    //wczytywanie danych z pliku
    if (file.is_open())
    {
        while (getline(file, line))
            content.push_back(line);
    }
    file.close();

    //sprawdzanie czy liczba zapisana w binarce jest podzielna przez dwa lub osiem
    //1. Liczba binarna jest podzielna przez dwa, jeśli na końcu znajduje się '0'
    //2. Liczba binarna jest podzielna przez osiem, jeśli trzy ostatnie znaki są zerami
    int dividedByTwo = 0, dividedByEight = 0;

    for (int i = 0; i < content.size(); i++)
    {
        if (content[i][content[i].size() - 1] == '0')
        {
            dividedByTwo++;

            if (content[i][content[i].size() - 2] == '0' and content[i][content[i].size() - 3] == '0')
                dividedByEight++;
        }
    }

    return "4.2. Liczb liczb podzielnych przez dwa: " + to_string(dividedByTwo) + "; Liczba liczb podzielnych przez osiem: " + to_string(dividedByEight);
}